pygmt/datatypes/__init__.py

```diff
@@ -0,0 +1,5 @@
+"""
+Wrappers for GMT data types.
+"""
+from pygmt.datatypes.dataset import _GMT_DATASET
+from pygmt.datatypes.grid import _GMT_GRID
```

Comment:
In #2398 and #2729, you have the `GMT_DATASET` and `GMT_GRID` classes in one file, but here, you've separated them into two files. I think this is ok, since we'll need to wrap [`GMT_IMAGE`](https://docs.generic-mapping-tools.org/6.5/devdocs/api.html#gmt-images) and [`GMT_CUBE`](https://docs.generic-mapping-tools.org/6.5/devdocs/api.html#gmt-cubes) at some point.